src/util.js

```diff
@@ -41,6 +41,21 @@ Util.isIFrame = function() {
   }
 };
 
+// Get a unique, incrementing Id value for any object on the page.
+Util.getUniqueId = (function(prefix) {
+  var i = Math.floor(Math.random() * 999) + 100;
+  var pfx = prefix || '';
+  function inc(pfx) {
+    if (!pfx) {
+      pfx = '';
+    } else {
+      pfx += '-';
+    }
+    return pfx + i++;
+  }
+  return inc;
+})();
```

Comment:
OK, will change and resubmit -pete

On Mon, Jan 11, 2016 at 9:42 PM, Boris Smus <notifications@github.com>
wrote:

> In src/util.js
> <https://github.com/borismus/webvr-boilerplate/pull/89#discussion_r49417654>
> :
>
> > @@ -41,6 +41,21 @@ Util.isIFrame = function() {
> >    }
> >  };
> >
> > +// Get a unique, incrementing Id value for any object on the page.
> > +Util.getUniqueId = (function(prefix) {
> > +  var i = Math.floor(Math.random() * 999) + 100;
> > +  var pfx = prefix || '';
> > +  function inc(pfx) {
> > +    if (!pfx) {
> > +      pfx = '';
> > +    } else {
> > +      pfx += '-';
> > +    }
> > +    return pfx + i++;
> > +  }
> > +  return inc;
> > +})();
>
> Another objection is to possible collisions. Could you switch to using
> UUIDs?
>
> http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/borismus/webvr-boilerplate/pull/89/files#r49417654>.
>



-- 
============================================================
=================
*Dr. Pete Markiewicz*
*Email:* pindiespace@gmail.com
*Sustainable virtual design blog:*
http://sustainablevirtualdesign.wordpress.com
*Sustainability template:* http://greenboilerplate.com
*Teaching site: *http://plyojump.com <http://www.plyojump.com/>
*Buy my book! *- Millennials and the Popular Culture
*Lifecourse:* http://www.lifecourse.com/store/catalog/lca/mpc.html
* AmazonAuthor Page:*
http://www.amazon.com/Pete-Markiewicz/e/B0069VZ7BG/ref=ntt_athr_dp_pel_pop_2
*On that page, see my DARPA Grand Challenge 2005 entry video, "Team
Robomonster":*
*Team Robomonster *(DARPA 2005 Grand Challenge):
http://robotsthatjump.wordpress.com/category/team-robomonster/
*Indiespace *(1st to sell downloaded music, March 1994):
http://www.indiespace.com. Also see http://archive.org under
"Kspace.com" 1996 for our original site.
============================================================
=================